src/main/integrations/discord-presence/index.ts

```diff
@@ -84,86 +79,61 @@ export default class DiscordPresence implements IIntegration {
   private ready = false;
   private connectionRetryTimeout: string | number | NodeJS.Timeout = null;
   private pauseTimeout: string | number | NodeJS.Timeout = null;
-  private endTimestamp: number | null = null;
   private stateCallback: (event: PlayerState) => void = null;
 
-  private lastTimeSeconds: number | null = null;
-  private lastDuration: number | null = null;
-
-  private lastEndTimestamp: number | null = null;
-  private lastVideoDetailsTitle: string | null = null;
-  private lastVideoDetailsAuthor: string | null = null;
-  private lastVideoDetailsId: string | null = null;
-  private lastTrackState: VideoState | null = null;
+  private videoDetails: VideoDetails | null = null;
 
   private connectionRetries: number = 0;
 
-  private playerStateChanged(state: PlayerState) {
-    if (this.ready && state.videoDetails) {
-      const date = Date.now();
-      const timeSeconds = Math.floor(date / 1000);
-      const videoProgress = Math.floor(state.videoProgress);
-      const duration = state.videoDetails.durationSeconds - videoProgress;
-
-      let adjustedTimeSeconds = timeSeconds;
-
-      if (!this.lastTimeSeconds) this.lastTimeSeconds = timeSeconds;
-      if (!this.lastDuration) this.lastDuration = duration;
-      if (timeSeconds - this.lastTimeSeconds > 0) {
-        if (this.lastDuration === duration) {
-          // The time changed seconds but the duration hasn't. We use the old timestamp to prevent weird deviations
-          adjustedTimeSeconds = this.lastTimeSeconds;
-        } else {
-          this.lastDuration = duration;
-          this.lastTimeSeconds = timeSeconds;
+  public listeningActivityType : boolean = true; // change outside
```

Comment:
Please do not use this (and the setting of this in the `main/index.ts`)
And this also forces the listening presences upon restart even if the user has requested for the Playing.

Please instead provide this integration with the settings provider as is done with other ones, and add a listener for the settings change for this to update this variable.
Additionally probably worth doing a forced SetActivity as well, to update this for the user for better UX